config.org



* Random stuff i dont understand
#+begin_src emacs-lisp

;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-

;; Place your private configuration here! Remember, you do not need to run 'doom
;; sync' after modifying this file!


;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Joel Green"
      user-mail-address "joelgreen737@gmail.com")

;(setq tramp-default-method "ssh")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:

#+end_src
*  Font setup
#+begin_src emacs-lisp
(setq doom-font (font-spec :family "SauceCodePro Nerd Font" :size 13 )
       doom-variable-pitch-font (font-spec :family "SauceCodePro Nerd Font" :size 13))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
#+end_src

* Setting theme
#+begin_src emacs-lisp
(setq doom-theme 'doom-monokai-pro)

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
#+end_src
* Setting up org directories
#+begin_src emacs-lisp
(setq org-directory "~/TestDev/org/")
(setq org-super-agenda "~/TestDev/org/")

#+end_src
* display line numbers
#+begin_src emacs-lisp
;; This determines the style of line numbers in effect. If set to `nil', line
;; numbers are disabled. For relative line numbers, set this to `relative'.
(setq display-line-numbers-type t)
#+end_src

* just modeline things
#+begin_src emacs-lisp
;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
;;
;;SauceCodePro Nerd Font
(display-time-mode 1)                             ; Enable time in the mode-line
(unless (equal "Battery status not available"
               (battery))
  (display-battery-mode 1))
#+end_src


* set scratch buffer defautl mode to org
#+begin_src emacs-lisp
(setq initial-major-mode 'org-mode)
#+end_src

* Coding related
#+begin_src emacs-lisp

(after! flycheck

  (setq flycheck-check-syntax-automatically '(mode-enabled save new-line idle-change)))




(setq lsp-clients-angular-language-server-command
      '("node"
        "/home/shaw/.npm-global/lib/node_modules/@angular/language-server"
        "--ngProbeLocations"
        "/home/shaw/.npm-global/lib/node_modules"
        "--tsProbeLocations"
        "/home/shaw/.npm-global/lib/node_modules"
        "--stdio"))


(setq undo-limit 80000000                         ; Raise undo-limit to 80Mb
      evil-want-fine-undo t)                      ; By default while in insert all changes are one big blob. Be more granular

;; eslint
(add-hook 'js2-mode-hook
          (defun my-js2-mode-setup ()
            (flycheck-mode t)
            (when (executable-find "eslint")
              (flycheck-select-checker 'javascript-eslint))))

(setq projectile-enable-caching nil)
#+end_src


config.el
;;; $DOOMDIR/config.el -*- lexical-binding: t; -*-

;; Place your private configuration here! Remember, you do not need to run 'doom
;; sync' after modifying this file!


;; Some functionality uses this to identify you, e.g. GPG configuration, email
;; clients, file templates and snippets.
(setq user-full-name "Joel Green"
      user-mail-address "joelgreen737@gmail.com")

;(setq tramp-default-method "ssh")

;; Doom exposes five (optional) variables for controlling fonts in Doom. Here
;; are the three important ones:
;;
;; + `doom-font'
;; + `doom-variable-pitch-font'
;; + `doom-big-font' -- used for `doom-big-font-mode'; use this for
;;   presentations or streaming.
;;
;; They all accept either a font-spec, font string ("Input Mono-12"), or xlfd
;; font string. You generally only need these two:
;; (setq doom-font (font-spec :family "monospace" :size 12 :weight 'semi-light)
;;       doom-variable-pitch-font (font-spec :family "sans" :size 13))

;; There are two ways to load a theme. Both assume the theme is installed and
;; available. You can either set `doom-theme' or manually load a theme with the
;; `load-theme' function. This is the default:
(setq doom-theme 'doom-monokai-pro)

;; If you use `org' and don't want your org files in the default location below,
;; change `org-directory'. It must be set before org loads!
(setq org-directory "~/TestDev/org/")
(setq org-super-agenda "~/TestDev/org/")

;; This determines the style of line numbers in effect. If set to `nil', line
;; numbers are disabled. For relative line numbers, set this to `relative'.
(setq display-line-numbers-type t)


;; Here are some additional functions/macros that could help you configure Doom:
;;
;; - `load!' for loading external *.el files relative to this one
;; - `use-package!' for configuring packages
;; - `after!' for running code after a package has loaded
;; - `add-load-path!' for adding directories to the `load-path', relative to
;;   this file. Emacs searches the `load-path' when you load packages with
;;   `require' or `use-package'.
;; - `map!' for binding new keys
;;
;; To get information about any of these functions/macros, move the cursor over
;; the highlighted symbol at press 'K' (non-evil users must press 'C-c c k').
;; This will open documentation for it, including demos of how they are used.
;;
;; You can also try 'gd' (or 'C-c c d') to jump to their definition and see how
;; they are implemented.
;;
(display-time-mode 1)                             ; Enable time in the mode-line

(unless (equal "Battery status not available"
               (battery))
  (display-battery-mode 1))


(after! flycheck

(setq flycheck-check-syntax-automatically '(mode-enabled save new-line idle-change)))




(setq lsp-clients-angular-language-server-command
  '("node"
    "/home/shaw/.npm-global/lib/node_modules/@angular/language-server"
    "--ngProbeLocations"
    "/home/shaw/.npm-global/lib/node_modules"
    "--tsProbeLocations"
    "/home/shaw/.npm-global/lib/node_modules"
    "--stdio"))


(setq undo-limit 80000000                         ; Raise undo-limit to 80Mb
      evil-want-fine-undo t)                      ; By default while in insert all changes are one big blob. Be more granular

;; eslint
(add-hook 'js2-mode-hook
          (defun my-js2-mode-setup ()
            (flycheck-mode t)
            (when (executable-find "eslint")
              (flycheck-select-checker 'javascript-eslint))))

(setq projectile-enable-caching nil)

(setq initial-major-mode 'org-mode)
